src/Tool/SqlQueryParser.php

```diff
@@ -0,0 +1,105 @@
+<?php declare(strict_types=1);
+
+namespace Manticoresearch\Buddy\Core\Tool;
+
+use PHPSQLParser\PHPSQLCreator;
+use PHPSQLParser\PHPSQLParser;
+
+class SqlQueryParser
+{
+
+	/**
+	 * @var SqlQueryParser|null
+	 */
+	protected static self|null $instance = null;
+	/**
+	 * @var array<string, array<int, array{table:string, expr_type:string, base_expr:string,
+	 *   sub_tree:array<int, array<string, string|bool>>}>>|null
+	 */
+	private static array|null $parsedPayload = null;
+	/**
+	 * @var PHPSQLParser
+	 */
+	private static PHPSQLParser $parser;
+	/**
+	 * @var PHPSQLCreator
+	 */
+	private static PHPSQLCreator $creator;
+
+
+	final private function __construct() {
+	}
+
+	final protected function __clone() {
+	}
+
+	final public function __wakeup() {
+		throw new \Exception('Cannot unserialize a singleton.');
+	}
+
+	/**
+	 * @return self
+	 */
+	public static function getInstance(): self {
+		if (self::$instance === null) {
+			self::$instance = new self();
+			self::$parser = new PHPSQLParser();
+			self::$creator = new PHPSQLCreator();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * @return PHPSQLParser
+	 */
+	private static function getParser(): PHPSQLParser {
+		return self::$parser;
+	}
+
+	/**
+	 * @return PHPSQLCreator
+	 */
+	private static function getCreator(): PHPSQLCreator {
+		return self::$creator;
+	}
+
+	/**
+	 * @param string $payload
+	 * @return array<string, array<int,
+	 *   array{table:string, expr_type:string, base_expr:string,
+	 * sub_tree: array<int, array<string, string|bool>>}>>|null
+	 */
+	public static function parse(string $payload): ?array {
+		$parsedPayload = static::getInstance()::getParser()->parse($payload);
+		static::setParsedPayload($parsedPayload);
+		return $parsedPayload;
+	}
+
+	/**
+	 * @return string
+	 * @throws \PHPSQLParser\exceptions\UnsupportedFeatureException
```

Comment:
We should avoid throwing exceptions that can't be handled outside of plugins. Instead, we should catch them and throw an internal exception. This way, external plugins can work with it if needed.

Reply:
Done